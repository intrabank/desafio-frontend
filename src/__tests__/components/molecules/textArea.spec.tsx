import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import TextArea from '@/components/molecules/textarea';
import { render, screen } from '@testing-library/react';

describe('TextArea', () => {
	it('should render textarea', () => {
		const tree = renderer
			.create(<TextArea label='Teste' htmlFor='teste' />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render textarea with error', () => {
		render(<TextArea label='Teste' htmlFor='teste' state='error' />);

		const textarea = screen.findByRole('textbox');
		const label = screen.getByText('Teste');

		expect(label.innerHTML).toBe('Teste<span> *</span>');
	});

	it('should render textArea with no error', () => {
		render(
			<TextArea
				label='Teste'
				placeholder='teste'
				htmlFor='teste'
				state='active'
			/>
		);

		const textarea = screen.getByPlaceholderText('teste');
		const label = screen.getByText('Teste');

		expect(textarea.getAttribute('aria-invalid')).toBe('false');
		expect(label.innerHTML).toBe('Teste');
	});

	it('should render error message', () => {
		render(
			<TextArea
				label='Teste'
				placeholder='teste'
				htmlFor='teste'
				error='Error message'
				state='error'
			/>
		);

		const errorMessage = screen.getByText('Error message');

		expect(errorMessage).toBeInTheDocument();
	});
});
