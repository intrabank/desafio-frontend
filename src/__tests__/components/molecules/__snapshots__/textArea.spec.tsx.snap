// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should render textarea 1`] = `
<div
  className="c-UFKVs"
>
  <label
    className="c-bUsXAP c-bUsXAP-fXZMCZ-state-idle"
    htmlFor="teste"
    onMouseDown={[Function]}
  >
    Teste
  </label>
  <textarea
    aria-invalid={false}
    className="c-gOvnqN c-gOvnqN-gFpoIP-state-idle"
  />
</div>
`;
